claat/fetch.go

```diff
@@ -227,22 +228,42 @@ func fetchDriveFile(id string, nometa bool) (*resource, error) {
 
 var crcTable = crc64.MakeTable(crc64.ECMA)
 
-func slurpBytes(client *http.Client, dir, url string, n int) (string, error) {
-	res, err := retryGet(client, url, n)
+func slurpBytes(client *http.Client, codelabSrc, dir, imgURL string, n int) (string, error) {
+	// images can be local in Markdown cases or remote.
+	// Only proceed a simple copy on local reference.
+	var b []byte
+	var ext string
+	u, err := url.Parse(imgURL)
 	if err != nil {
 		return "", err
 	}
-	defer res.Body.Close()
-	b, err := ioutil.ReadAll(res.Body)
+	if u.Host == "" {
+		imgURL = path.Join(path.Dir(codelabSrc), imgURL)
```

Comment:
how about prepending "/" to imgURL here, to make sure it doesn't go like "../../../etc/passwd" or any other sensitive content.

then, path.Join should take care of cleaning up the final result anyway.